src/rviz/selection/selection_handler.cpp

```diff
@@ -214,7 +215,7 @@ void SelectionHandler::destroyBox(const std::pair<CollObjectHandle, uint64_t>& h
     Ogre::WireBoundingBox* box = it->second.second;
 
     node->detachAllObjects();
-    node->getParentSceneNode()->removeAndDestroyChild(node->getName());
+    if(node->getName().length() > 0) node->getParentSceneNode()->removeAndDestroyChild(node->getName());
```

Comment:
Won't this leak anonymous nodes?

In other places I replaced `removeAndDestroyChild(node->getName())` with:
```c++
node->getParentSceneNode()->removeChild(node);
node->getCreator()->destroySceneNode(node);
```

Reply:
Thanks for the suggestion. Will test.